### DISCUSSION TITLE
Multi-Fidelity Models with Variable Design Spaces

### DISCUSSION THREAD

**Discussion by socrates-ferna** [SMTorg/smt#598] [Ideas]
Hi, thanks for such an amazing toolbox!

I have read both the 2.0 and the original release papers looking for a way of building a multifidelity model in which the design space of the low fidelity model has less dimensions than the high fidelity space. This does not seem to be possible with the current Co-Kriging model. The new Hierarchical Design Spaces, which can tackle variable design spaces, seem to be designed only for single-fidelity models.  Across the literature, I have only found the technique on design space mappings to deal with this case.

I can think of a very naive way of dealing with this, which is expanding the low-fid input vectors onto the high-fid space forcing the low fid points to be constant along the new dimensions, but I was wondering if I could find a better solution within the toolbox.

An aerospace example of my case would be a low fid surrogate of lift and drag coefficients that only depends on the angle of attack, and a high fid surrogate which accounts in addition for Reynolds and Mach.

Is there any tool in SMT for my issue? If there's not, is it in the roadmap?

Thanks again for your work!

**Comment by relf** [comment]
Hi thanks for the kind words! So far MFK family models work with nested doe and same dimension space for hi and lo-fi. As you guessed, you may find some inspiring building blocks in SMT (MFK, designspace) but it has to be implemented. 

Regarding the roadmap, to the best of my knowledge, we have some plans to remove the nested doe condition on MFK but still with the same dimension.

**Reply by socrates-ferna** [reply]
Thanks for you answer! If you happen to know about a reference or paper in which I could find a proposal to solve this case, I would be very grateful if you could link it here. I would be happy to attempt an implementation and possibly try to integrate it in the toolbox. If there's nothing on your radar on this topic, you can consider the discussion closed.

Best regards!
